lib/classes/Swift/Transport/MailTransport.php

```diff
@@ -247,9 +247,32 @@ private function _getReversePath(Swift_Mime_Message $message)
     private function _formatExtraParams($extraParams, $reversePath)
     {
         if (false !== strpos($extraParams, '-f%s')) {
-            $extraParams = empty($reversePath) ? str_replace('-f%s', '', $extraParams) : sprintf($extraParams, escapeshellarg($reversePath));
+            $extraParams = empty($reversePath) ? str_replace('-f%s', '', $extraParams) : sprintf($extraParams, $this->_escapeCommand($reversePath));
         }
 
         return !empty($extraParams) ? $extraParams : null;
     }
-}
+
+    /**
+     * Escapes a string using either escapeshellarg or wrapping the command in single quotes if the function
+     * has been disabled.
+     *
+     * @param string $command
+     *
+     * @return string
+     */
+    private function _escapeCommand($command)
+    {
+        if (is_string($command)) {
```

Comment:
Are you referencing the param type within the docblock? I was being defensive to make sure a string was passed.